flag.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"flag"
+	"os"
+
+	jujutesting "github.com/juju/testing"
+)
+
+var noTestMongoJs *bool = flag.Bool("notest-mongojs", false, "Disable MongoDB tests that require JavaScript")
+
+func init() {
+	if os.Getenv("JUJU_NOTEST_MONGOJS") == "1" || jujutesting.MgoServer.WithoutV8 {
```

Comment:
I think let's just use either the env var or the flag. Probably the env var 'cos that means you can go test ./... .